R/Task.R

```diff
@@ -800,6 +799,7 @@ Task = R6Class("Task",
         private$.internal_valid_task = NULL
         return(invisible(private$.internal_valid_task))
       }
+      private$.row_hash = NULL
```

Comment:
the row hash is only with respect to the primary task, right? 
If so, we do not need to reset it here.

Reply:
The function also changes the primary task. But I see that I am doing it twice currently. Line 809 changes it already to NULL when the pimary task is changed.